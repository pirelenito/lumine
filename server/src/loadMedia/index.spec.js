const path = require('path')
const temp = require('temp')
const promisify = require('util').promisify
const mkdirTemp = promisify(temp.mkdir)
const loadMedia = require('.')

it('load a RAW photo', async () => {
  jest.setTimeout(30000)

  const cacheFolder = await mkdirTemp('testing')
  const source = path.join(__dirname, '../../spec/fixtures/DSC02482.ARW')

  const media = await loadMedia(cacheFolder)(source)
  expect(media).toMatchSnapshot()
})
