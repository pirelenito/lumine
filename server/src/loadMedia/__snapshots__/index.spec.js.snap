// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load a RAW photo 1`] = `
Object {
  "id": "4bff511b810cba0590e0d787232e9bda2deb4039",
  "metadata": Object {
    "exif": Object {
      "BrightnessValue": -2.8265625,
      "ColorSpace": 1,
      "CompressedBitsPerPixel": 8,
      "Contrast": 0,
      "CreateDate": "2016:12:19 23:27:24",
      "CustomRendered": 0,
      "DateTimeOriginal": "2016:12:19 23:27:24",
      "DigitalZoomRatio": 1,
      "ExifImageHeight": 4000,
      "ExifImageWidth": 6000,
      "ExposureCompensation": 0,
      "ExposureMode": 0,
      "ExposureProgram": 3,
      "ExposureTime": 0.625,
      "FNumber": 2.2,
      "Flash": 16,
      "FocalLength": 35,
      "FocalLengthIn35mmFormat": 52,
      "ISO": 200,
      "InteropOffset": 38114,
      "LensModel": "E 35mm F1.8 OSS",
      "LightSource": 11,
      "MaxApertureValue": 1.6953125,
      "MeteringMode": 5,
      "RecommendedExposureIndex": 200,
      "Saturation": 0,
      "SceneCaptureType": 0,
      "SensitivityType": 2,
      "Sharpness": 0,
      "WhiteBalance": 1,
    },
    "gps": Object {},
    "image": Object {
      "ExifOffset": 348,
      "ImageDescription": "                               ",
      "Make": "SONY",
      "Model": "ILCE-6000",
      "ModifyDate": "2016:12:19 23:27:24",
      "Orientation": 1,
      "ProcessingSoftware": "UFRaw 0.20",
      "Software": "ILCE-6000 v1.00",
      "SubfileType": 1,
      "YCbCrPositioning": 2,
    },
  },
  "resources": Object {
    "preview": "/tmp/testing118021-1110-s2ny1k.zbfpp/preview/4b/ff511b810cba0590e0d787232e9bda2deb4039.jpg",
    "source": "/usr/src/app/server/spec/fixtures/DSC02482.ARW",
    "thumbnail": "/tmp/testing118021-1110-s2ny1k.zbfpp/thumbnail/4b/ff511b810cba0590e0d787232e9bda2deb4039.jpg",
  },
}
`;
